homeassistant/components/mysensors/config_flow.py

```diff
@@ -0,0 +1,176 @@
+                    asyncio.create_task(gw.stop())
+        except Exception as e:
+            _LOGGER.info("try_connect failed with exception", exc_info=e)
+            return False
+
+
+class MySensorsConfigFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow."""
+
+    async def async_step_import(self, user_input):
+        """Import a config entry.
+
+        Special type of import, we're not actually going to store any data.
+        Instead, we're going to rely on the values that are in config file.
+        """
+        return await self.async_step_user(user_input=user_input)
+
+    async def async_step_user(self, user_input=None):
+        """Create a config entry from frontend user input."""
+        errors = {}
+        if user_input is not None:
+            is_mqtt = user_input[CONF_DEVICE] == MQTT_COMPONENT
+            is_serial = False
+            try:
+                if user_input[CONF_DEVICE] != MQTT_COMPONENT:
+                    try:
+                        await self.hass.async_add_executor_job(
+                            is_serial_port, user_input[CONF_DEVICE]
+                        )
+                        is_serial = True
+                    except vol.Invalid:
+                        await self.hass.async_add_executor_job(
+                            is_socket_address, user_input[CONF_DEVICE]
+                        )
+                        is_serial = False
+            except vol.Invalid:
+                errors[CONF_DEVICE] = "invalid_device"
+
+            try:
+                if is_mqtt and CONF_TOPIC_IN_PREFIX in user_input:
+                    valid_subscribe_topic(user_input[CONF_TOPIC_IN_PREFIX])
+            except vol.Invalid:
+                errors[CONF_TOPIC_IN_PREFIX] = "invalid_subscribe_topic"
+            try:
+                if is_mqtt and CONF_TOPIC_OUT_PREFIX in user_input:
+                    valid_publish_topic(user_input[CONF_TOPIC_OUT_PREFIX])
+            except vol.Invalid:
+                errors[CONF_TOPIC_OUT_PREFIX] = "invalid_publish_topic"
+            if not is_mqtt and (
+                CONF_TCP_PORT in user_input
+                and (user_input[CONF_TCP_PORT] < 1 or user_input[CONF_TCP_PORT] > 65535)
+            ):
+                errors[CONF_TCP_PORT] = "invalid_port"
+            try:
+                if CONF_PERSISTENCE_FILE in user_input:
+                    is_persistence_file(user_input[CONF_PERSISTENCE_FILE])
+            except vol.Invalid:
+                errors[CONF_TCP_PORT] = "invalid_persistence_file"
+
+            uniquestr = user_input[CONF_DEVICE]
+            if is_mqtt:
+                uniquestr += (
+                    user_input[CONF_TOPIC_IN_PREFIX] + user_input[CONF_TOPIC_OUT_PREFIX]
+                )
+            elif not is_serial:
+                uniquestr += str(user_input[CONF_TCP_PORT])
+            gateway_id = hashlib.sha256(uniquestr.encode()).hexdigest()[:8]
```

Comment:
what "entry id" do you mean? hopefully not `id()` of the ConfigEntry?

Reply:
The `ConfigEntry.entry_id` that is available after the entry is created.

Reply:
Okay, but what about persistence file naming? The previous scheme allowed the user to delete and re-create an entry and it would load the already existing persistence file. I found that quite useful, altough maybe that was just because i was debugging config flows

Reply:
Yeah, that's a good point. I think we can keep the persistence file option, but make it optional. If it doesn't exist we generate a default name, eg based on the config entry entry_id in the integration setup.

Reply:
Ok, i have added it. Ideally i would like to hide it behind some sort of advanced toggle, but i dont think that is supported in config flows.